app/components/select_all_child_component.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SelectAllChildComponent < ActionView::Component::Base
+  def initialize(type:, key:, name: nil)
+    @name = name
+    @key = key
+    @type = type
+  end
+
+  def final_name
+    @name ||= @type
+
+    "#{@name}_list[]"
+  end
+
+  private
+
+    attr_reader :name, :type, :key
```

Comment:
So I guess the main reason to do this is to avoid calling instance variables directly. I like it. I should refactor my existing components because I don't think I included it.

Reply:
Oh I actually got the idea from you here https://github.com/psu-libraries/myaccount/blob/475904401417f8783f47b6de746d1179153bf4a6/app/components/bib_item_component.rb#L30

Reply:
`:type` here refers to a bib record's type, like `BOOK` or `ATLAS` as opposed to `hold` or `checkout` which would be a model, right?

Reply:
I see ok, yeah, in dashboard item I didn't. Will make that change 👍